nuxeo-core/nuxeo-core-convert/src/test/java/org/nuxeo/ecm/core/convert/tests/TestCache.java

```diff
@@ -94,4 +100,33 @@ public void testCache() throws Exception {
         assertEquals(0, cacheSize1 - cacheSize3);
     }
 
+    @Test
+    public void shouldNotReturnNullFileCacheEntries() throws IOException {
+        File file = FileUtils.getResourceFileFromContext("test-data/hello.doc");
+        assertNotNull(file);
+        assertTrue(file.length() > 0);
+
+        // Put a blobholder in the cache
+        Blob blob = Blobs.createBlob(file, "application/msword", null, "hello.doc");
```

Comment:
You can probably simplify to the one-parameter constructor, the other ones shouldn't be needed.